token-metadata/program/tests/utils/programmable.rs

```diff
@@ -17,62 +18,117 @@ use solana_sdk::{
+static PROGRAM_ALLOW_LIST: [Pubkey; 1] = [mpl_token_auth_rules::ID];
+
+macro_rules! get_primitive_rules {
+    (
+        $source_owned_by_sys_program:ident,
+        $dest_program_allow_list:ident,
+        $dest_pda_match:ident,
+        $source_program_allow_list:ident,
+        $source_pda_match:ident,
+        $dest_owned_by_sys_program:ident,
+        $nft_amount:ident,
+    ) => {
+        let $source_owned_by_sys_program = Rule::ProgramOwned {
+            program: system_program::ID,
+            field: PayloadKey::Source.to_string(),
+        };
+
+        let $dest_program_allow_list = Rule::ProgramOwnedList {
+            programs: PROGRAM_ALLOW_LIST.to_vec(),
+            field: PayloadKey::Destination.to_string(),
+        };
+
+        let $dest_pda_match = Rule::PDAMatch {
+            program: None,
+            pda_field: PayloadKey::Destination.to_string(),
+            seeds_field: PayloadKey::DestinationSeeds.to_string(),
+        };
+
+        let $source_program_allow_list = Rule::ProgramOwnedList {
+            programs: PROGRAM_ALLOW_LIST.to_vec(),
+            field: PayloadKey::Source.to_string(),
+        };
+
+        let $source_pda_match = Rule::PDAMatch {
+            program: None,
+            pda_field: PayloadKey::Source.to_string(),
+            seeds_field: PayloadKey::SourceSeeds.to_string(),
+        };
+
+        let $dest_owned_by_sys_program = Rule::ProgramOwned {
+            program: system_program::ID,
+            field: PayloadKey::Destination.to_string(),
+        };
+        let $nft_amount = Rule::Amount {
+            field: PayloadKey::Amount.to_string(),
+            amount: 1,
+            operator: CompareOp::Eq,
+        };
+    };
+}
+
+pub async fn create_default_metaplex_rule_set(
     context: &mut ProgramTestContext,
-    payer: Keypair,
-    name: String,
+    creator: Keypair,
 ) -> (Pubkey, AuthorizationData) {
-    // --------------------------------
-    // Create RuleSet
-    // --------------------------------
-    // Find RuleSet PDA.
+    let name = String::from("Metaplex Royalty Enforcement");
     let (ruleset_addr, _ruleset_bump) =
-        mpl_token_auth_rules::pda::find_rule_set_address(payer.pubkey(), name.clone());
+        mpl_token_auth_rules::pda::find_rule_set_address(creator.pubkey(), name.clone());
+
+    get_primitive_rules!(
+        source_owned_by_sys_program,
+        dest_program_allow_list,
+        dest_pda_match,
+        source_program_allow_list,
+        source_pda_match,
+        dest_owned_by_sys_program,
+        nft_amount,
+    );
 
-    // Rule for Transfers: Allow transfers to a Token Owned Escrow account.
-    let owned_by_token_metadata = Rule::ProgramOwned {
-        program: mpl_token_metadata::ID,
-        field: PayloadKey::Destination.to_string(),
+    // (source is owned by system program && dest is on allow list && destination is a PDA) ||
+    // (source is on allow list && source is a PDA && dest is owned by system program)
```

Comment:
This does not seem like the RuleSet you want for default as it limits dest to PDA only.   That was like a draft, but I have since updated it.  Do you want the new one that is?:
```
// amount is 1 &&
// (source is owned by system program || (source is on allow list && source is a PDA) &&
// (dest is owned by system program || (dest is on allow list && dest is a PDA)
```

See this update:
https://github.com/metaplex-foundation/mpl-token-auth-rules/blob/main/program/tests/basic_royalty_enforcement.rs#L88-L115

Reply:
I think my text just doesn't match what the actual rule is. If you look at the code the rule should be what you describe in your text above.